server/msgpack/server.go

```diff
@@ -40,159 +37,62 @@ const (
 	unknownRemoteHostAddress = "<unknown>"
 )
 
-type serverMetrics struct {
-	openConnections tally.Gauge
+// NewServer creates a new msgpack server.
+func NewServer(address string, aggregator aggregator.Aggregator, opts Options) xserver.Server {
+	iOpts := opts.InstrumentOptions()
+	scope := iOpts.MetricsScope()
+	return xserver.NewServer(
+		address,
+		NewHandler(
```

Comment:
It might be easier to read to pull these out into
```
instrumentOpts := opts.InstrumentOptions()
scope := instrumentOpts.MetricsScope()

handlerScope := scope.Tagged(map[string]string{"handler": "msgpack"})
handlerInstrumentOpts := opts.SetInstrumentOptions(iOpts.SetmetricsScope(handlerScope))
handler := NewHandler(aggregator, handlerInstrumentOpts)

serverScope := scope.Tagged(map[string]string{"server": "msgpack"})
serverInstrumentOpts := instrumentOpts.SetMetricsScope(serverScope)
serverOpts := xserver.NewOptions().
  SetInstrumentOptions(serverInstrumentOpts).
  SetRetryOptions(opts.RetryOptions())

return xserver.NewServer(address, handler, serverOpts)
```